src/lib/rnp.c

```diff
@@ -1200,11 +1200,116 @@ rnp_decrypt_file(rnp_ctx_t *ctx, const char *f, const char *out)
+ *  @param newpath preallocated pointer which will store the result on success
+ *  @param maxlen maximum number of chars in newfile, including the trailing \0
+ *  @param overwrite whether it is allowed to overwrite output file by default
+ *  @return true on success, or false otherwise (user cancels the operation)
+ **/
+
+static bool
+rnp_get_output_filename(const char *path, char *newpath, size_t maxlen, bool overwrite)
+{
+    char reply[10];
+
+    if (strlen(path) == 0) {
+        fprintf(stdout, "Please enter the filename: ");
+        if (fgets(newpath, maxlen, stdin) == NULL) {
+            return false;
+        }
+        rnp_strip_eol(newpath);
+    } else {
+        strncpy(newpath, path, maxlen);
+    }
+
+    while (true) {
+        if (rnp_file_exists(newpath)) {
+            if (overwrite) {
+                unlink(newpath);
+                return true;
+            }
+
+            fprintf(stdout,
+                    "File '%s' already exists. Would you like to overwrite it (y/N)?",
+                    newpath);
+
+            if (fgets(reply, sizeof(reply), stdin) == NULL) {
+                return false;
+            }
+            if (strlen(reply) > 0 && toupper(reply[0]) == 'Y') {
+                unlink(newpath);
+                return true;
+            }
+
+            fprintf(stdout, "Please enter the new filename: ");
+            if (fgets(newpath, maxlen, stdin) == NULL) {
+                return false;
+            }
+
+            rnp_strip_eol(newpath);
+
+            if (strlen(newpath) == 0) {
+                return false;
+            }
+        } else {
+            return true;
+        }
+    }
+}
+
+/** @brief Initialize input for streamed RNP operation, based on input filename/path
+ *  @param ctx Initialized RNP operation context
+ *  @param src Allocated source structure to put result in
+ *  @param in Input filename/path. For NULL or '-' stdin source will be created
+ *  @return true on success, or false otherwise
+ **/
+
+static bool
+rnp_initialize_input(rnp_ctx_t *ctx, pgp_source_t *src, const char *in)
+{
+    rnp_result_t res;
+    bool         is_stdin = in == NULL || in[0] == '\0' || strcmp(in, "-") == 0;
+
+    if (is_stdin) {
+        res = init_stdin_src(src);
+    } else {
+        res = init_file_src(src, in);
+    }
+
+    return res == RNP_SUCCESS;
+}
+
+/** @brief Initialize output for streamed RNP operation, based on output filename/path.
+ *         Will commuicate with user if file already exists and overwrite flag is not set.
+ *  @param ctx Initialized RNP operation context
+ *  @param dst Allocated dest structure to put result in
+ *  @param out Output filename/path as provided by user. For NULL or '-' stdout will be used.
+ *  @return true on success or false otherwise
+ **/
+
+static bool
+rnp_initialize_output(rnp_ctx_t *ctx, pgp_dest_t *dst, const char *out)
+{
+    char newname[PATH_MAX];
+    bool is_stdout = out == NULL || out[0] == '\0' || strcmp(out, "-") == 0;
```

Comment:
can you put () around those conditions. It hard to read this way

Reply:
Ok.